File: android/src/main/java/com/nitroinputmask/engine/MaskEngineProtocol.kt
package com.nitroinputmask.engine

interface MaskEngineProtocol {
  fun apply(input: String): Pair<String, String>
  val wantsTrailingCursor: Boolean
  /** Expanded mask string for cursor positioning. Null means put cursor at end. */
  val expandedMask: String?
}

File: ios/engine/MaskEngineProtocol.swift
protocol MaskEngineProtocol {
  func apply(input: String) -> (masked: String, raw: String)
  var wantsTrailingCursor: Bool { get }
  /// Expanded mask string used for cursor positioning (e.g. "999.999.999-99").
  /// Return nil if cursor should just go to end of masked string.
  var expandedMask: String? { get }
}
